### PR TITLE
Tweak rpc logging settings

### PR DESCRIPTION
**Pull request by drewbanin**
- respect logs_start parameter even when tasks have completed
- fix for incorrect hook index for on-run-end hooks
- clean up on-run hook log messages
- include SQL run in the context of a hook in the hook log messages

### CODE REVIEW

File: core/dbt/task/run.py [hunk 1]
```diff
@@ -125,25 +125,23 @@ def run_hooks(self, adapter, hook_type: RunHookType, extra_context):
 
             hook_text = '{}.{}.{}'.format(hook.package_name, hook_type,
                                           hook.index)
-            print_hook_start_line(hook_text, idx, num_hooks)
-
             hook_meta_ctx = HookMetadata(hook, self.index_offset(idx))
             uid_ctx = UniqueID(hook.unique_id)
-            with uid_ctx, hook_meta_ctx, startctx:
-                logger.debug('on-run-hook starting')
+            running_ctx = DbtModelState({'node_status': 'running'})
```
Comment (beckjake) [inline core/dbt/task/run.py]: Would it make more sense to put the entire rest of this method under a `with uid_ctx:` and tab everything in?

Comment (drewbanin) [reply core/dbt/task/run.py]: yep, that's a really good idea!

**Comment by beckjake** [comment]
@drewbanin I can't request review from you, but I think this is all set

File: core/dbt/task/run.py [hunk 2]
```diff
@@ -125,25 +125,24 @@ def run_hooks(self, adapter, hook_type: RunHookType, extra_context):
...
-            print_hook_start_line(hook_text, idx, num_hooks)
-
             hook_meta_ctx = HookMetadata(hook, self.index_offset(idx))
-            uid_ctx = UniqueID(hook.unique_id)
-            with uid_ctx, hook_meta_ctx, startctx:
-                logger.debug('on-run-hook starting')
-            status = 'OK'
+            running_ctx = DbtModelState({'node_status': 'running'})
+            with UniqueID(hook.unique_id):
+                with hook_meta_ctx, startctx, running_ctx:
```
Comment (drewbanin) [inline core/dbt/task/run.py]: We actually don't want to use the `running_ctx` here - I had to check how we do this for other "regular" types of nodes - that status is only emitted at the end of the resource's execution. Can we just remove the `running_ctx` from this line (and the definition above)?

Sorry for the confusion there! This one was my bad.

Comment (drewbanin) [reply core/dbt/task/run.py]: actually i'll just push a fix for this haha

**Pull request merged**